jax/_src/lax/linalg.py

```diff
@@ -694,8 +704,14 @@ def _triangular_solve_gpu_translation_rule(
     return xops.TriangularSolve(a, b, left_side, lower, unit_diagonal,
                                 transpose)
 
-xla.backend_specific_translations['gpu'][triangular_solve_p] = \
-    _triangular_solve_gpu_translation_rule
+if cusolver is not None:
+  xla.backend_specific_translations['gpu'][triangular_solve_p] = \
+      partial(_triangular_solve_gpu_translation_rule, cusolver.trsm)
+
+if rocsolver is not None:
+  # TODO (inailuig) remove always_use_gpu_kernel=True once complex trsm is added to XLA
```

Comment:
Unless I'm remembering wrong, the XLA triangular solve already fully supports complex arguments?

Reply:
Not yet on ROCm (see: https://github.com/tensorflow/tensorflow/blob/6859f52a3fba6714b5360262f190c9649613ac5c/tensorflow/stream_executor/rocm/rocm_blas.cc#L2432). I think @deven-amd might already be working on adding it (https://github.com/google/jax/issues/2012#issuecomment-742764789)

Reply:
https://github.com/ROCmSoftwarePlatform/tensorflow-upstream/pull/1203/commits/3ceb49c107977b31a5a6dc7552886695ff82d9c8

Reply:
this change is now in the ROCm fork ( https://github.com/ROCmSoftwarePlatform/tensorflow-upstream/pull/1203 ) and 
a PR has been created to upstream it ( https://github.com/tensorflow/tensorflow/pull/45697 )